lxd/instance/drivers/driver_qemu.go

```diff
@@ -7075,6 +7075,48 @@ func (d *qemu) checkFeatures(hostArch int, qemuPath string) (map[string]any, err
 		features["cpu_hotplug"] = struct{}{}
 	}
 
+	// Check AMD SEV features (only for x86 architecture)
+	if hostArch == osarch.ARCH_64BIT_INTEL_X86 {
+		cmdline, err := os.ReadFile("/proc/cmdline")
+		if err != nil {
+			return nil, err
+		}
+
+		parts := strings.Split(string(cmdline), " ")
+
+		// Check if SME is enabled in the kernel command line.
+		if shared.StringInSlice("mem_encrypt=on", parts) {
+			features["sme"] = struct{}{}
+		}
+
+		// Check if SEV/SEV-ES are enabled
+		sev, err := os.ReadFile("/sys/module/kvm_amd/parameters/sev")
+		if err != nil && !os.IsNotExist(err) {
+			return nil, err
+		} else if string(sev) == "Y\n" {
+			// Host supports SEV, check if QEMU supports it as well.
+			capabilities, err := monitor.SEVCapabilities()
+			if err != nil {
+				logger.Debug("Failed querying SEV capability during VM feature check", logger.Ctx{"err": err})
+			} else {
+				features["sev"] = struct{}{}
```

Comment:
I was thinking we should actually store the `capabilities` var in `features["sev"]`, rather than split the "sev" feature into multiple sub-features, when really its just properties of the "sev" feature.